ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+                                            '--product', fields_in_filenames[i], \
+                                            '--no-add_vdate_subdir'])
+                            result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
+                            rc = result.returncode
+
+                        # The raw file name needs to be the same as what the retrieve_data.py
+                        # script called above ends up retrieving.  The list of possibile templates
+                        # for this name is given in parm/data_locations.yml, but which of those
+                        # is actually used is not known until retrieve_data.py completes.  Thus,
+                        # that information needs to be passed back by the script and used here.
+                        # For now, we hard-code the file name here.
+                        if obtype == 'CCPA':
+                            hr = yyyymmddhh.hour
+                            fn_raw = 'ccpa.t' + f'{hr:02d}' + 'z.' + accum_obs_formatted + 'h.hrap.conus.gb2'
+                        elif obtype == 'NOHRSC':
+                            yyyymmddhh_str = dt.datetime.strftime(yyyymmddhh, '%Y%m%d%H')
+                            fn_raw = 'sfav2_CONUS_' + accum_obs_formatted + 'h_' + yyyymmddhh_str + '_grid184.grb2'
+                        elif obtype == 'MRMS':
+                            hr = yyyymmddhh.hour
+                            fn_raw = fields_in_filenames[i] + '_' + levels_in_filenames[i] \
+                                   + '_' + yyyymmdd_task_str + '-' + f'{hr:02d}' + '0000.grib2'
+                            fn_raw = os.path.join('topofhour', fn_raw)
+                        elif obtype == 'NDAS':
+                            time_ago = yyyymmddhh_arcv - yyyymmddhh
+                            hrs_ago = int(time_ago.seconds/3600)
+                            hh_arcv_str = dt.datetime.strftime(yyyymmddhh_arcv, '%H')
+                            fn_raw = 'nam.t' + hh_arcv_str + 'z.prepbufr.tm' + f'{hrs_ago:02d}' + '.nr'
+                        fp_raw = os.path.join(arcv_dir_raw, fn_raw)
+
+                        # Get the full path to the final processed obs file (fp_proc) we want to
+                        # create.
+                        indx = obs_retrieve_times_crnt_day.index(yyyymmddhh)
+                        fp_proc = all_fp_proc_dict[fg][indx]
+
+                        # Make sure the directory in which the processed file will be created exists.
+                        dir_proc = os.path.dirname(fp_proc)
+                        Path(dir_proc).mkdir(parents=True, exist_ok=True)
+
+                        msg = dedent(f"""
+                            Creating the processed obs file
+                                {fp_proc}
+                            from the raw file
+                                {fp_raw}
+                            ...
+                            """)
+                        logging.info(msg)
```

Comment:
low priority message
```suggestion
                        logging.debug(msg)
```

Reply:
Done.